examples/fidelity.py

```diff
@@ -0,0 +1,86 @@
+"""Simulates the fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  # Builds an arbitrary circuit to test. The circuit is non Clifford to show the
+  # use of simulators.
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def simulate_trace(circuit, pauli_gates, qubits, noise):
+  simulator = cirq.DensityMatrixSimulator(noise=noise)
+
+  n = len(pauli_gates)
+  d = 2**n
+
+  trace = 0
+  for x in range(d):
+    xbin = numpy.binary_repr(x, width=n)
+
+    pauli_string = cirq.PauliString(dict(zip(qubits, pauli_gates)))
+    display = cirq.approx_pauli_string_expectation(pauli_string, num_samples=n)
+    y = cirq.sample(cirq.Circuit(
+        pauli_string,
```

Comment:
why are you adding pauli_string to a circuit?

Reply:
I couldn't find any way to input the initial state in cirq.sample() and thus did the pre-rotation. I remembered that you had told me not to do it.

In some previous iteration, reviewer Craig Gidney (Strilanc) asked me not to use the DenseMatrixSimulator which allowed to input a different initial state, and thus not do the pre-rotation.

I had left that comment unresolved and I will leave this one unresolved too, since I suspect you are both in agreement and I just don't understand what both of you are telling me.